gnuradio-runtime/include/gnuradio/basic_block.h

```diff
@@ -71,6 +72,11 @@ namespace gr {
 
     gr::thread::mutex mutex;          //< protects all vars
 
+    /*! Used by blocks to access the logger system.
+     */
+    gr::logger_ptr d_logger;
+    gr::logger_ptr d_debug_logger;
+
```

Comment:
These duplicate the existing definitions in block.h.  Since basic_block is the parent class of block.h, having them here will allow them to be inherited by class block, so they should be removed there.  Also, in order for this to work, these members should be in the protected section of the class.definition.